### PR TITLE
Use native RSA/OpenSSL crypto whenever possible

### PR DESCRIPTION
**Pull request by connor4312**
This improves the performance of cracking certain jwks by 15x.

### CODE REVIEW

File: lib/algorithms/rsassa.js [hunk 1]
```diff
@@ -60,7 +60,21 @@ function rsassaV15SignFn(name) {
     return promise;
   };
 
-  return helpers.setupFallback(null, webcrypto, fallback);
+  var node;
```
Comment (linuxwolf) [inline lib/algorithms/rsassa.js]: this variable should be named `nodejs` for consistency.

File: lib/algorithms/rsassa.js [hunk 2]
```diff
@@ -118,7 +132,26 @@ function rsassaV15VerifyFn(name) {
...
```
Comment (linuxwolf) [inline lib/algorithms/rsassa.js]: this variable should be named `nodejs` for consistency.

File: lib/algorithms/rsa-util.js [hunk 3]
```diff
@@ -49,7 +50,25 @@ function convertToJWK(key, isPublic) {
   return result;
 }
 
+var pemCache = new WeakMap();
```
Comment (linuxwolf) [inline lib/algorithms/rsa-util.js]: This code base still runs on platforms where WeakMap is not available.

Personally, I see caching as a nice-to-have, so maybe forgo it for now in favor of a method of caching it in the key representation ...

**Comment by connor4312** [comment]
Re caching: it seems quite beneficial to cache the pem, conversion from the default key format to pem is not free (particularly since it generally has to be done on every request)

```
     111,313,315 op/s » with caching
           4,771 op/s » without caching
```

I swapped it for a caching mechanism using a non-enumerable property via Object.defineProperty. This is the strategy taken in several weakmap polyfills and works well in any ES5-capable environment.

Let me know what you think 😄

**Comment by linuxwolf** [comment]
👍
I might refactor the naming some in the future, but this works for me!

Thanks!

**Pull request merged**